internal/plugin/connectors/http/generic/oauth/v1/protocol_test.go

```diff
@@ -0,0 +1,352 @@
+				Host:     "example.com",
+				Path:     "/wp-json/wp/v2/posts",
+				RawQuery: "include_entities=true",
+			},
+			Header: map[string][]string{
+				"Authorization": {"doesn't matter"},
+				"Content-Type":  {"application/x-www-form-urlencoded"},
+			},
+			Body: createRequestBody("status=Hello%20Ladies%20%2B%20Gentlemen%2C%20a%20signed%20OAuth%20request%21"),
+		},
+	}
+
+	result := collectParameters(params.oauth1, params.r)
+
+	want := map[string][]string{
+		"include_entities":       {"true"},
+		"oauth_consumer_key":     {"xvz1evFS4wEEPTGEFPHBog"},
+		"oauth_nonce":            {"kYjzVBB8Y0ZFabxSWbWovY3uYSQ2pTgmZeNu2VS4cg"},
+		"oauth_signature_method": {"HMAC-SHA1"},
+		"oauth_timestamp":        {"1318622958"},
+		"oauth_token":            {"370773112-GmHxMAgYyLbNEtIKZeRNFsMKPR9EyMZeS9weJAEb"},
+		"oauth_version":          {"1.0"},
+		"status":                 {"Hello%20Ladies%20%2B%20Gentlemen%2C%20a%20signed%20OAuth%20request%21"},
+	}
+
+	if reflect.DeepEqual(result, want) {
+		t.Log("Passed collectParameters()!")
+	} else {
+		t.Errorf("collectParameters() FAILED, expected %v but got %v", want, result)
+	}
+}
+
+func Test_generateParameterString(t *testing.T) {
+	type args struct {
+		paramMap map[string][]string
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    string
+		wantErr bool
+	}{
+		{
+			name: "Can generate a valid Parameter String with no body or query params",
+			args: args{
+				paramMap: map[string][]string{
+					"oauth_consumer_key":     {"key"},
+					"oauth_nonce":            {"nonce"},
+					"oauth_signature_method": {"HMAC-SHA1"},
+					"oauth_timestamp":        {"123456789"},
+					"oauth_token":            {"token"},
+					"oauth_version":          {"1.0"},
+				},
+			},
+			want: "oauth_consumer_key=key&oauth_nonce=nonce&oauth_signature_method=HMAC-SHA1&oauth_timestamp=123456789&oauth_token=token&oauth_version=1.0",
+		},
+		{
+			name: "Can generate a valid Parameter String with no body or query params",
+			args: args{
+				paramMap: map[string][]string{
+					"oauth_consumer_key":     {"key"},
+					"oauth_nonce":            {"nonce"},
+					"oauth_signature_method": {"HMAC-SHA1"},
+					"oauth_timestamp":        {"123456789"},
+					"oauth_token":            {"token"},
+					"oauth_version":          {"1.0"},
+				},
+			},
+			want: "oauth_consumer_key=key&oauth_nonce=nonce&oauth_signature_method=HMAC-SHA1&oauth_timestamp=123456789&oauth_token=token&oauth_version=1.0",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := generateParameterString(tt.args.paramMap); got != tt.want {
+				t.Errorf("generateParameterString() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_generateBaseString(t *testing.T) {
+	type args struct {
+		paramString string
+		r           *gohttp.Request
+	}
+	tests := []struct {
+		name string
+		args args
+		want string
+	}{
+		{
+			name: "Can generate a valid Base string: All components supplied",
```

Comment:
When you switch to using assert, it'll print what the name of the test is (i.e. Test_generateBaseString) which means you can strike "Can generate a valid Base string", as the context will be readable by just the method name. Which is a sign of good naming conventions! 👍